js/w3c/headers.js

```diff
@@ -155,6 +156,18 @@ define(
                     ret += ", <span class='ed_mailto'><a class='u-email email' " + rm + " href='mailto:" + p.mailto + "'>" + p.mailto + "</a></span>";
                 }
                 if (p.note) ret += " (" + p.note + ")";
+                if (p.extras && p.extras.length !== 0) {
```

Comment:
It's best to use the declarative form... avoid nested stuff, and the string concatenation. 

```JS
var resultHTML = p.extras
  // Remove empty names
  .filter(function (extra) {
    return extra.name.trim();
  })
  // Convert to HTML
  .map(function(extra) {
    var span = doc.createElement('span');
    var textContainer = span;
    if (extra.class) {
      span.class = extra.class;
    }
    if (extra.href) {
      var a = doc.createElement('a');
      span.appendChild(a);
      a.href = extra.href;
      textContainer = a;
    }
    textContainer.innerHTML = extra.name;
    return span.outerHTML;
  })
  .join(', ');
```